### PR TITLE
Arm64 support

### PR DESCRIPTION
**Pull request by Erikvv**
Replace build.sh script by Makefile (god help us) so part of it can be run on a different platform or container.

The new release contains 2 binaries, one for each of the supported platforms.
    
I tested this on a Dragonboard 410c.

Helps https://github.com/storj-thirdparty/nextcloud-app/issues/2#issuecomment-940014795

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -30,17 +30,36 @@ pipeline {
...
+                always {
+                    cleanWs()
+                }
+            }
+        }
+        stage('Go build arm64') {
+            agent {
+                dockerfile {
+                    dir 'docker/go-docker'
+                    args '--user root:root --volume /var/run/docker.sock:/var/run/docker.sock --volume /tmp/gomod:/go/pkg/mod '
```
Comment (ifraixedes) [inline Jenkinsfile:56]: I remember that in the past @stefanbenten was worried about exposing the host machine docker daemon inside of a container.

Comment (Erikvv) [reply Jenkinsfile:56]: In general the Jenkins setup allows for PR authors to abuse the Jenkins nodes.

Exposing the docker socket adds a vector when the container is compromised but that seems a lot less likely.

Keeping build/test logic out of the Jenkinsfile is very appealing to me. I actually also hope to use docker-in-docker to simplify the stage "PHPUnit" (in the future).

I also made #19 which I hope improves security should a container get compromised.

Comment (ifraixedes) [reply Jenkinsfile:56]: If it's fine for @stefanbenten it's fine for me.

**Comment by Erikvv** [comment]
Is merged

**Pull request closed without merging**